### DISCUSSION TITLE
How to format Ruby using rubyfmt?

### DISCUSSION THREAD

**Discussion by rphf** [zed-industries/zed#13669] [Feature Requests]
Hello,

I'm trying to set up the Zed to automatically format Ruby files using `rubyfmt` upon saving.
Here is the configuration I currently have in my `settings.json`:

```json
{
  "Ruby": {
    "format_on_save": {
      "external": {
        "command": "rubyfmt",
        "arguments": ["-i", "--", "{buffer_path}"]
      }
    }
  }
}
```

This configuration isn't working. I'm uncertain if the `{buffer_path}` argument is correctly recognized in this context, even though it works fine with the `prettier` command.  For more details on rubyfmt CLI specs, see its [github readme](https://github.com/penelopezone/rubyfmt).

If anyone has successfully configured rubyfmt with Zed, could you share your configuration or point out what might be going wrong with mine?

Thank you!

**Comment by kindjar** [comment]
FWIW, here's what I use:
```json
      "format_on_save": {
        "external": {
          "command": "rubyfmt",
          "arguments": ["--fail-fast"]
        }
      }
```

**Comment by schpet** [comment]
i'm able to do this via ruby-lsp

```json
{
  "languages": {
    "Ruby": {
      "language_servers": ["ruby-lsp", "!solargraph", "!rubocop"],
      "formatter": "language_server"
    }
  },
  "lsp": {
    "ruby-lsp": {
      "initialization_options": {
        "enabledFeatures": {
          "diagnostics": false
        },
        "formatter": "rubyfmt"
      },
      "settings": {
        "use_bundler": true
      }
    }
  }
}
```

it also requires this in the Gemfile:

```rb
group :development do
  gem "ruby-lsp-rubyfmt", git: "https://github.com/aaronlifton/ruby-lsp-rubyfmt.git", branch: "main"
end
```

references:
- https://github.com/jscharf/ruby-lsp-rubyfmt/pull/11
- https://github.com/schpet/dotfiles/blob/702dae114aab3e98dac14361420acb3f211fecc4/.config/zed/settings.json#L68-L85

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
